app/controllers/manageBusinesses/manage/CheckYourAnswersController.scala

```diff
@@ -0,0 +1,248 @@
+    Logger("application").error("[CheckYourAnswersController]" + errorMessage)
+    (if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler).showInternalServerError()
+  }
+
+
+  def submit(isAgent: Boolean, incomeSourceType: IncomeSourceType): Action[AnyContent] = auth.authenticatedAction(isAgent) { implicit user =>
+
+    val successCall = getSuccessCall(isAgent, incomeSourceType)
+    val errorCall = getErrorCall(incomeSourceType, isAgent)
+
+    withIncomeSourcesFS {
+      withSessionData(JourneyType(Manage, incomeSourceType), journeyState = BeforeSubmissionPage) { sessionData =>
+        sessionData.manageIncomeSourceData.map(x => (x.taxYear, x.reportingMethod, x.incomeSourceId)) match {
+          case Some((Some(taxYear), Some(reportingMethod), incomeSourceIdStringOpt)) =>
+            val incomeSourceBusinessName: Option[String] = user.incomeSources.getIncomeSourceBusinessName(incomeSourceType, incomeSourceIdStringOpt)
+            val incomeSourceIdOpt = incomeSourceIdStringOpt.map(id => IncomeSourceId(id))
+            val incomeSourceId: Option[IncomeSourceId] = user.incomeSources.getIncomeSourceId(incomeSourceType, incomeSourceIdOpt.map(m => m.value))
+
+            handleSubmitRequest(errorCall, isAgent, successCall, TaxYear(startYear = taxYear - 1, endYear = taxYear), incomeSourceId, ReportingMethod(reportingMethod), incomeSourceBusinessName, incomeSourceType)
+          case _ => Future.successful(logAndShowError(isAgent, s"[handleSubmitRequest]: Missing session values"))
+        }
+      }
+    }
+  }
+
+  private def handleSubmitRequest(errorCall: Call,
+                                  isAgent: Boolean,
+                                  successCall: Call,
+                                  taxYear: TaxYear,
+                                  incomeSourceIdMaybe: Option[IncomeSourceId],
+                                  reportingMethod: ReportingMethod,
+                                  incomeSourceBusinessName: Option[String],
+                                  incomeSourceType: IncomeSourceType
+                                 )(implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
+
+    val updateIncomeSourceResFuture = updateIncomeSource(taxYear.endYear, incomeSourceIdMaybe, reportingMethod)
+
+    val resultFuture = updateIncomeSourceResFuture flatMap {
+      case _: UpdateIncomeSourceResponseError =>
+        handleUpdateError(errorCall, isAgent, reportingMethod, taxYear, incomeSourceBusinessName, incomeSourceType)
+      case _: UpdateIncomeSourceResponseModel =>
+        incomeSourceIdMaybe match {
+          case Some(incomeSourceId) =>
+            handleSuccessfulUpdate(errorCall, successCall, isAgent, reportingMethod, taxYear, incomeSourceId, incomeSourceBusinessName, incomeSourceType)
+          case _ => Future.failed(MissingSessionKey(ManageIncomeSourceData.incomeSourceIdField))
+        }
+    }
+
```

Comment:
we usually not splitting statement returning Future and recover, as there will be a temptation to put some code in between in the future